From 74bca662e9a47d05d61fd7868a55036d1f4220ee Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 30 Aug 2015 06:30:52 +0300
Subject: [PATCH 0389/1240] fix: tools: fixed memory test in SPL

- Memory test access address 0, and it issue an exception
- This patch change the dummy parameters to be on the stack to let
  the test use it

Change-Id: Id37912d2c1b858bd39c62f80eaf99d786deda8f4
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23316
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/cpu/mvebu-common/tools/ddr_test.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm/cpu/mvebu-common/tools/ddr_test.c b/arch/arm/cpu/mvebu-common/tools/ddr_test.c
index 32af0dd..ce28b3b 100644
--- a/arch/arm/cpu/mvebu-common/tools/ddr_test.c
+++ b/arch/arm/cpu/mvebu-common/tools/ddr_test.c
@@ -233,7 +233,7 @@ static ulong mem_test_alt(vu_long *buf, ulong start_addr, ulong end_addr, vu_lon
 int run_memory_test(void)
 {
 	ulong start, end;
-	vu_long *buf, *dummy;
+	vu_long *buf, dummy;
 	int iteration_limit;
 	int ret;
 	ulong errs = 0;	/* number of errors */
@@ -247,13 +247,12 @@ int run_memory_test(void)
 	debug("%s:%d: start %#08lx end %#08lx\n", __func__, __LINE__, start, end);
 
 	buf = map_sysmem(start, end - start);
-	dummy = map_sysmem(0, sizeof(vu_long));
 	for (iteration = 0;
 			!iteration_limit || iteration < iteration_limit;
 			iteration++) {
 		printf("Iteration: %6d\r", iteration + 1);
 		debug("\n");
-		errs += mem_test_alt(buf, start, end, dummy);
+		errs += mem_test_alt(buf, start, end, &dummy);
 	}
 
 	printf("Tested %d iteration(s) with %lu errors.\n", iteration, errs);
-- 
1.9.1

